src/chains/mainnet/vm/precompiles.ts

```diff
@@ -91,60 +92,280 @@ export const precompiles: Precompile[] = [
+      },
+    ],
+    output: [
+      {
+        byteStart: 0,
+        byteLength: 32,
+        name: 'x',
+        description: "X coordinate of the result point on the elliptic curve 'alt_bn128'",
+      },
+      {
+        byteStart: 32,
+        byteLength: 32,
+        name: 'y',
+        description: "Y coordinate of the result point on the elliptic curve 'alt_bn128'",
+      },
+    ],
+    references: ['https://eips.ethereum.org/EIPS/eip-196'],
   },
   {
     address: '0x0000000000000000000000000000000000000007',
     name: 'ecMul',
     description: "Scalar multiplication (MUL) on the elliptic curve 'alt_bn128'",
     minGas: 6000,
-    input: [],
-    output: [],
-    references: [],
+    input: [
+      {
+        byteStart: 0,
+        byteLength: 32,
+        name: 'x1',
+        description: "X coordinate of the first point on the elliptic curve 'alt_bn128'",
+      },
+      {
+        byteStart: 32,
+        byteLength: 32,
+        name: 'y1',
+        description: "Y coordinate of the first point on the elliptic curve 'alt_bn128'",
+      },
+      {
+        byteStart: 64,
+        byteLength: 32,
+        name: 's',
+        description: 'Scalar to use for the multiplication',
+      },
+    ],
+    output: [
+      {
+        byteStart: 0,
+        byteLength: 32,
+        name: 'x',
+        description: "X coordinate of the result point on the elliptic curve 'alt_bn128'",
+      },
+      {
+        byteStart: 32,
+        byteLength: 32,
+        name: 'y',
+        description: "Y coordinate of the result point on the elliptic curve 'alt_bn128'",
+      },
+    ],
+    references: ['https://eips.ethereum.org/EIPS/eip-196'],
   },
   {
     address: '0x0000000000000000000000000000000000000008',
     name: 'ecPairing',
     description: "Bilinear function on groups on the elliptic curve 'alt_bn128'",
     minGas: 45000,
-    input: [],
-    output: [],
-    references: [],
+    input: [
+      {
+        byteStart: 0,
+        byteLength: 32,
+        name: 'x1',
+        description: "X coordinate of the first point on the elliptic curve 'alt_bn128'",
+      },
+      {
+        byteStart: 32,
+        byteLength: 32,
+        name: 'y1',
+        description: "Y coordinate of the first point on the elliptic curve 'alt_bn128'",
+      },
+      {
+        byteStart: 64,
+        byteLength: 32,
+        name: 'x3',
+        description: "X coordinate of the third point on the elliptic curve 'alt_bn128'",
+      },
+      {
+        byteStart: 96,
+        byteLength: 32,
+        name: 'x2',
+        description: "X coordinate of the second point on the elliptic curve 'alt_bn128'",
+      },
+      {
+        byteStart: 128,
+        byteLength: 32,
+        name: 'y3',
+        description: "Y coordinate of the third point on the elliptic curve 'alt_bn128'",
+      },
+      {
+        byteStart: 160,
+        byteLength: 32,
+        name: 'y2',
```

Comment:
I think the order got swapped here (it goes x1,y1,x3,x2,y3,y2). Let's keep the first two as `x1`,`y1` and `x2,y2`, then make the last one `xn`,`yn` to indicate it's arbitrary length number of inputs